### ISSUE TITLE
Add requirements to README

### ISSUE THREAD

**Issue by thesuperzapper** [kubeflow/manifests#1842]
Currently it's unclear what the contract is between this repo and users, I think a "requirements" section in the README would help users understand and use the manifests.

__The main versions we should list are:__
- Min/Max kubernetes Version
- Min/Max kustomize Version
- Min/Max Istio Version
- Min/Max Cert-Manager Version
- Min/Max Argo-Workflow Version

__We should also list any other "requirements", for example:__
* the presence of a default StorageClass
* etc

**Comment by thesuperzapper** [comment]
@kubeflow/wg-manifests-leads thoughts?

**Comment by titansmc** [comment]
Hi,
Not sure if this is the right place to ask, but is the usage of a default StorageClass supported ?  I see the PVC have pending state because no StorageClass is being used.

**Comment by yanniszark** [comment]
@thesuperzapper thanks for opening an issue. Here are some thoughts:

> Min/Max kubernetes Version

We have validated on 1.17, but other versions should work as well.

> Min/Max kustomize Version

We have validated on 3.2.0 and encountered problem moving to v4: https://github.com/kubeflow/manifests/issues/1797

> Min/Max Istio Version

We have validated 1.9.0 (included in manifests). For a range, I would delegate to each individual component.

> Min/Max Cert-Manager Version

We have validate 0.11. For a range, I would delegate to each individual component.

> Min/Max Argo-Workflow Version

I would delegate to Pipelines.

> the presence of a default StorageClass

Yes, we should clarify that if there is a chance users bump into it.
We need a default StorageClass.

**Comment by thesuperzapper** [comment]
@yanniszark While putting this information in this issue is nice, I think we need to add it to the README.

**Comment by yanniszark** [comment]
Of course. Since you asked for some thoughts, I thought I would provide them :)
I will make a PR later today or tomorrow, if there are no more discussions here.

**Comment by Bobgy** [comment]
I think to properly support requirements in manifests, we need each WG to document their own requirements + compatibility matrix. Otherwise, it's hard for manifest WG to figure out.
